mmrotate/core/visualization/image.py

```diff
@@ -1,107 +1,244 @@
 # Copyright (c) OpenMMLab. All rights reserved.
-import math
 
 import cv2
+import matplotlib.pyplot as plt
+import mmcv
 import numpy as np
-from mmcv.image import imread, imwrite
-from mmcv.visualization.color import color_val
+from matplotlib.collections import PatchCollection
+from matplotlib.patches import Polygon
+from mmdet.core.visualization import palette_val
+from mmdet.core.visualization.image import draw_labels, draw_masks
 
+from mmrotate.core.visualization.palette import get_palette
 
-def imshow(img, win_name='', wait_time=0):
-    """Show an image.
+EPS = 1e-2
+
+
+def _get_adaptive_scales(areas, min_area=800, max_area=30000):
+    """Get adaptive scales according to areas.
+
+    The scale range is [0.5, 1.0]. When the area is less than
+    ``'min_area'``, the scale is 0.5 while the area is larger than
+    ``'max_area'``, the scale is 1.0.
 
     Args:
-        img (str or ndarray): The image to be displayed.
-        win_name (str): The window name.
-        wait_time (int): Value of waitKey param.
+        areas (ndarray): The areas of bboxes or masks with the
+            shape of (n, ).
+        min_area (int): Lower bound areas for adaptive scales.
+            Default: 800.
+        max_area (int): Upper bound areas for adaptive scales.
+            Default: 30000.
+
+    Returns:
+        ndarray: The adaotive scales with the shape of (n, ).
     """
-    cv2.imshow(win_name, imread(img))
-    if wait_time == 0:  # prevent from hanging if windows was closed
-        while True:
-            ret = cv2.waitKey(1)
+    scales = 0.5 + (areas - min_area) / (max_area - min_area)
+    scales = np.clip(scales, 0.5, 1.0)
+    return scales
 
-            closed = cv2.getWindowProperty(win_name, cv2.WND_PROP_VISIBLE) < 1
-            # if user closed window or if some key pressed
-            if closed or ret != -1:
-                break
-    else:
-        ret = cv2.waitKey(wait_time)
+
+def draw_bboxes(ax, bboxes, color='g', alpha=0.8, thickness=2):
+    """Draw bounding boxes on the axes.
+
```

Comment:
Can change the name to `draw_rbboxes` to make differences with mmdet.

Reply:
fixed